bigframes/core/transpiler/googlesql/expression.py

```diff
@@ -0,0 +1,82 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
```

Comment:
I would like to see a docstring for this module.

One important piece of information is how we should come up with names for these classes. Are we following: https://cloud.google.com/bigquery/docs/reference/standard-sql/query-syntax ? If so, I don't see alias_expression anywhere, only as_alias.

Alternatively, are we using the names from https://github.com/google/zetasql/blob/master/docs/functions-and-operators.md ?

Reply:
Updated the docstring. Yes, we are trying to follow GoogleSQL syntax document as much as possible. The ZetaSQL operators does not apply for us because we are not planning to introduce the type to the column expression so far. Besides, the `expression` is added with some extensions classes for convenience. One of extension is `alias expression`, which is close to `as_alias` so `as_alias` is omitted.  I have documented this special case in our codes.